scripts/generators/safe_struct_generator.py

```diff
@@ -903,12 +908,18 @@ def qfi_construct(item, member):
             construct_txt = ''      # Body of constuctor as well as body of initialize() func following init_func_txt
             destruct_txt = ''
 
+            has_pnext = struct.sType is not None
             copy_pnext = ''
+            copy_pnext_if = ''
             copy_strings = ''
             for member in struct.members:
                 m_type = member.type
                 if member.name == 'pNext':
-                    copy_pnext = '    pNext = SafePnextCopy(in_struct->pNext, copy_state);\n'
+                    copy_pnext = 'pNext = SafePnextCopy(in_struct->pNext, copy_state);\n'
+                    copy_pnext_if = f'''
```

Comment:
nit: you don't need `f`

```suggestion
                    copy_pnext_if = '''
                        if (copy_pnext) {
                            pNext = SafePnextCopy(in_struct->pNext, copy_state);
                        }'''
```